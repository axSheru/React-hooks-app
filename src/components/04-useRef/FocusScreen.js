import React, { useRef } from 'react';
import '../../index.css';

export const FocusScreen = () => {

    const inputRef = useRef();

    const handleClick = () => {
        inputRef.current.focus();
    };

    return (
        <div>
            <h1>FocusScreen</h1>
            <hr />

            <input
                type="text"
                ref={ inputRef }
                className='form-control'
                placeholder='Su nombre'
            />

            <button
                className='btn btn-outline-primary mt-3'
                onClick={ handleClick }
            >
                Focus
            </button>
        </div>
    );
}
