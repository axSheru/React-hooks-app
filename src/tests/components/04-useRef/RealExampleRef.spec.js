import { shallow } from "enzyme";
import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef";

describe('Pruebas en el componente RealExampleRef.', () => {

    test('debe de hacer match con el snapshot.', () => {

        const wrapper = shallow( <RealExampleRef /> );

        expect( wrapper ).toMatchSnapshot();

    });

});