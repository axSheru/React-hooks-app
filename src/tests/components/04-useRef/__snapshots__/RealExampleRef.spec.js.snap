// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente RealExampleRef. debe de hacer match con el snapshot. 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-outline-primary mt-5"
    onClick={[Function]}
  >
    Hide/Show
  </button>
</div>
`;
